### PR TITLE
added a version check so that users get feedback if a older version than tf 1.7 is installed

### PR DESCRIPTION
**Pull request by renepickhardt**
This should fix issues #70 and #66 (or at least give users a clearer feedback of what they need to do)

### CODE REVIEW

**Comment by jfkirk** [comment]
Thank you, @renepickhardt ! This is a great idea.

Looks like Flake8 is unhappy:
```tensorrec/tensorrec.py:59:1: E101 indentation contains mixed spaces and tabs
tensorrec/tensorrec.py:59:1: W191 indentation contains tabs
tensorrec/tensorrec.py:60:1: E101 indentation contains mixed spaces and tabs
tensorrec/tensorrec.py:60:14: E231 missing whitespace after ','
tensorrec/tensorrec.py:60:20: E231 missing whitespace after ','
tensorrec/tensorrec.py:62:78: W291 trailing whitespace
tensorrec/tensorrec.py:65:8: E101 indentation contains mixed spaces and tabs
tensorrec/tensorrec.py:65:8: W191 indentation contains tabs```

File: tensorrec/tensorrec.py [hunk 1]
```diff
@@ -56,7 +56,13 @@ def __init__(self,
         If True, a bias value will be calculated for every user feature and item feature.
         """
 
-        # Arg-check
+	# check TensorFlow version
+        major,minor,patch = tf.__version__.split(".")
+        if int(major) < 1 or int(major) == 1 and int(minor) < 7:
+            raise RuntimeError("""You need to have at least TensorFlow version 1.4 installed in order to use
```
Comment (jfkirk) [inline tensorrec/tensorrec.py]: Shouldn't the string say `version 1.7`?

Comment (renepickhardt) [reply tensorrec/tensorrec.py]: sorry my bad! I made the fix for version 1.4 and then when I realized that 1.7. is needed I forgot to update the string :( 
I will also fix the intendations. sorry for bad patch

Comment (jfkirk) [reply tensorrec/tensorrec.py]: Ah I just saw your Issue re: version 1.4 adding the `data` module. 

We also use `Iterator`s whose attributes changed in 1.7, so anything under 1.7 will break.

Comment (renepickhardt) [reply tensorrec/tensorrec.py]: well as u see in the if statement I ask for 1.7 it is just the string that was still 1.4. 
but what about the whitespace? my editor shows me the whitespace as spaces not tabs. but CI seems to fail. I will send another pullrequest in a minute.

**Comment by jfkirk** [comment]
Merged -- thank you @renepickhardt ! This makes you the first contributor to TensorRec that I didn't personally rope in to helping, and I'm grateful for it.

**Comment by renepickhardt** [comment]
just checked out your software this day and it seems to work like a charm. Not entirely sure if I can use it out of the box in the setting where I am planning to. But if yes, this will save me much time and effort. Therefor: thank YOU and not me for a minor patch.

Btw just saw you work at spotify this is funny since I wrote my diploma thesis 9 years ago about recommending music to people (At that time I have been running a small niche social networking site for fans and bands in the genre of heavy metal. We had user-band and user-record relations (including reviews and ratings)). Seeing what is possible nowadays just reminded me on how "cute" the efforts in my diploma thesis (to recommend simlar but less known artists) have been.

**Comment by jfkirk** [comment]
I appreciate the kind words, and it'd be very cool if TensorRec turns out to be useful to your setting -- let me know if I can help.

For what it's worth, this was what my thesis looked like: https://drive.google.com/file/d/1arAOpy_Zyh5bVICuf2wzTvHB-yGb4-Jo/view?usp=sharing

**Comment by renepickhardt** [comment]
omg that video made my day! (I guess the robot was supposed to stand?) My thesis is in German and has not visible output (the network is down by now) so sorry can't share anything back other than a funny story: 
I remember dating a girl in the old days and I knew three bands she liked and on a date I turned on music from my self calculated recommendations based on the three bands I knew she liked. It was a very funny moment when during the evening she pointed out that my taste of music was great. though I directly replied that it was her taste of music and how I knew that.

I will give you feedback once I know if tensorrec proves to be helpful in my setting (probably in the form of a blog article). and thanks for your offer I might come back to that.

**Pull request merged**